### PR TITLE
Add a text tokens api

### PR DESCRIPTION
**Pull request by ByMartrixx**
While the api part is finished, the implementation still needs work, particularly with implementing tokenization in every place it should be used. It also needs more tests. The option `dtt` (Dump Text Tokens) was added for debugging/testing purposes

As for the api: It's visitor based, and programs may add visitors with `TextTokenVisitor.addVisitor`.

### CODE REVIEW

File: src/org/jetbrains/java/decompiler/main/ClassWriter.java [hunk 1]
```diff
@@ -204,6 +205,8 @@ public void classLambdaToJava(ClassNode node, TextBuffer buffer, Exprent method_
           }
           buffer.append(" ->");
 
+          // @martrixx: left here!
```
Comment (SuperCoder7979) [inline src/org/jetbrains/java/decompiler/main/ClassWriter.java:208]: Debug comment left over?

File: src/org/jetbrains/java/decompiler/main/extern/IFernflowerPreferences.java [hunk 2]
```diff
@@ -258,6 +258,10 @@ public interface IFernflowerPreferences {
   @Description("Forces the processing of JSR instructions even if the class files shouldn't contain it (Java 7+)")
   String FORCE_JSR_INLINE = "fji";
 
+  @Name("Dump Text Tokens")
+  @Description("Dump Text Tokens on each class file")
+  String DUMP_TEXT_TOKENS = "dtt";
```
Comment (SuperCoder7979) [inline src/org/jetbrains/java/decompiler/main/extern/IFernflowerPreferences.java:263]: Should also add "dtt" to readme

File: src/org/jetbrains/java/decompiler/main/extern/TextTokenVisitor.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+  @SuppressWarnings("unchecked")
+  private static List<Factory> getFactories() {
+    List<Factory> property = (List<Factory>) DecompilerContext.getProperty(PROPERTY_NAME);
+    if (property == null) {
+      property = new ArrayList<>();
+      DecompilerContext.setProperty(PROPERTY_NAME, property);
+    }
+    return property;
```
Comment (SuperCoder7979) [inline src/org/jetbrains/java/decompiler/main/extern/TextTokenVisitor.java:33]: space after if statement

**Review by SuperCoder7979 — COMMENTED** [review]
Great starting point! Only minor nitpicks from me 😄

File: src/org/jetbrains/java/decompiler/main/ClassWriter.java (hunk 1 above)
Comment (ByMartrixx) [reply src/org/jetbrains/java/decompiler/main/ClassWriter.java:208]: Oh, this is just for me to keep track of from where I should continue working. I'll remove it in the next commit (and may add another one somewhere else, if I don't finish)

**Review by SuperCoder7979 — APPROVED** [review]

**Pull request merged**